### ISSUE TITLE
compiling on Mac

### ISSUE THREAD

**Issue by cariann09** [bbuchfink/diamond#181]
I am supplying this issue as it may be easy to fix a few warnings so that Diamond runs correctly on a mac. I was able to make diamond on a mac (10.9.5) with only a few warnings as listed below. I compared the results with Linux, and almost all results were the same except that some results differed by 3, sometimes making a Mac index <1,  e.g. tab results

Linux: subject query 82.1	134	23	1	402	1	10	142	7.9e-60	222.2
Mac  : subject query 82.1	134	23	1	399	-2	10	142	7.9e-60	222.2

Linux: subject query 76.7	133	23	3	549	163	1	129	2.2e-50	191.4
Mac  : subject query 76.7	133	23	3	546	160	1	129	2.2e-50	191.4

Warnings on Mac, but not on Linux.
sh build_simple.sh
src/output/join_blocks.cpp:155:2: warning: control may reach end of non-void function [-Wreturn-type]
        }
        ^
1 warning generated.
src/output/blast_tab_format.cpp:192:13: warning: enumeration values 'op_frameshift_forward' and 'op_frameshift_reverse' not handled in switch
      [-Wswitch]
                                switch (i.op()) {
                                        ^
1 warning generated.
src/output/sam_format.cpp:25:11: warning: enumeration values 'op_frameshift_forward' and 'op_frameshift_reverse' not handled in switch [-Wswitch]
                switch (i->op) {

**Comment by bbuchfink** [comment]
It seems to me like you are not using the latest software version on Mac.
Could yould please try to download the source code again and compile to
make sure.

**Comment by cariann09** [comment]
I downloaded it on Dec 10, diamond v0.9.13.114.

I just downloaded another and it is v0.9.14.115 -- a new release!
Only one warning:
sh build_simple.sh
src/output/join_blocks.cpp:155:2: warning: control may reach end of
non-void function [-Wreturn-type]

And, it gives the exact same results as from Linux! Great!

I apologize for not checking release dates before creating the 'issue' -
but its an issue re-solved ;)
Thank you.
Cari
